yang/ietf-schedule.yang

```diff
@@ -84,273 +96,414 @@ module ietf-schedule {
-          leaf period-duration {
-            type duration {
-              pattern 'P((([0-9]+)D)?(T(0[0-9]|1[0-9]|2[0-3])'
-                    + ':[0-5][0-9]:[0-5][0-9]))|P([0-9]+)W';
-            }
-            description
-              "A positive duration of the time. This value is 
-               equivalent to the format of duration type except that 
-               the value cannot be negative.";
-          }
+            "A positive duration of the time. This value is
+             equivalent to the format of duration type except that
+             the value cannot be negative.";
         }
       }
     }
   }
 
   grouping recurrence {
     description
-      "This grouping is defined to identify properties that contain a 
-       recurrence rule.";
-    reference 
-      "RFC 5545: Internet Calendaring and Scheduling Core Object 
-       Specification (iCalendar), Section 3.3.10";        
-    container recurrence {
+      "A simple definition of recurrence.";
+    container recurrence-first {
       description
-        "Recurrence rule definition.";
-      leaf freq {
-        type enumeration {
-          enum secondly {
-            value 1;
-            description
-              "Repeating events based on an interval of a second
-               or more.";
-          }
-          enum minutely {
-            value 2;
-            description
-              "Repeating events based on an interval of a minute
-               or more.";
-          }
-          enum hourly {
-            value 3;
-            description
-              "Repeating events based on an interval of an hour
-               or more.";
-          }
-          enum daily {
-            value 4;
-            description
-              "Repeating events based on an interval of a day or
-               more.";
-          }
-          enum weekly {
-            value 5;
-            description
-              "Repeating events based on an interval of a week or
-               more.";
-          }
-          enum monthly {
-            value 6;
-            description
-              "Repeating events based on an interval of a month or
-               more.";
-          }
-          enum yearly {
-            value 7;
-            description
-              "Repeating events based on an interval of a year or
-               more.";
-          }
+        "Specifies the first instance of the recurrence.";
+      leaf date-time-start {
+        type union {
+          type yang:date-no-zone;
+          type yang:date-and-time;
         }
-        mandatory true;
         description
-          "This parameter is defined to identify the type of
-           recurrence rule.";
+          "Defines the first instance in the recurrence set. If it is
+           specified as a date-no-zone value type with no duration
+           specified, the recurrence's duration is taken to be one
+           day. It SHOULD match the pattern of the recurrence rule,
```

Comment:
When the "SHOULD" can be safely ignored?

Reply:
RFC 5545 states that: The "DTSTART"
      property value SHOULD match the pattern of the recurrence rule, if
      specified.  The recurrence set generated with a "DTSTART" property
      value that doesn't match the pattern of the rule is undefined.

In most cases, the date and time of the first instance determines the date and time of following instances of recurrence set. But I think in some other cases like a recurrence date (not a date-and-time, e.g., every last working day of each month or anniversary) is defined in the recurrenece rule, it is okay for a date-time-start to not fit in the pattern of the recurrence.
But anyway, this parameter is always defined as the first instance in the recurrence set.

Reply:
I would remove the SHOULD from the description + add a ref statement the appropriate section in 5545

Reply:
sure, updated.